src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.DotNet.ApiSymbolExtensions;
+
+namespace Microsoft.DotNet.GenAPI
+{
+    /// <summary>
+    /// Processes assemly symbols to build correspoding structures in C# language.
+    /// Depend on ISyntaxWriter impelemtenation, the result could be C# source file, xml etc.
+    /// </summary>
+    internal class CSharpFileBuilder : IAssemblySymbolWriter, IDisposable
+    {
+        private readonly TextWriter _textWriter;
+        private readonly ISymbolFilter _symbolFilter;
+        private readonly CSharpSyntaxRewriter _syntaxRewriter;
+
+        private readonly AdhocWorkspace _adhocWorkspace;
+        private readonly SyntaxGenerator _syntaxGenerator;
+
+
+        public CSharpFileBuilder(ISymbolFilter symbolFilter, TextWriter textWriter, CSharpSyntaxRewriter syntaxRewriter)
+        {
+            _textWriter = textWriter;
+            _symbolFilter = symbolFilter;
+            _syntaxRewriter = syntaxRewriter;
+
+            _adhocWorkspace = new AdhocWorkspace();
+            _syntaxGenerator = SyntaxGenerator.GetGenerator(_adhocWorkspace, LanguageNames.CSharp);
+        }
+
+        public void WriteAssembly(IAssemblySymbol assembly) => Visit(assembly);
```

Comment:
Add `<inheritdoc />`

Reply:
Added. Plus, added comment to base class.